### DISCUSSION TITLE
Can I export in a "diary" format?

### DISCUSSION THREAD

**Discussion by boldandbusted** [xwmx/nb#258] [Q&A]
Thanks for this amazing project! :)

A current project I'm on is where I'm acting as a "user persona", and I'm writing something like a "Dear Diary" set of notes. I'd like to create a unified single-file output (PDF & Markdown primary outputs, HTML, maybe LibreOffice formats) that has all of my individual notes concatenated, with metadata headers (timestamp, maybe others?) wrapping the content, like:

(Rough draft - just for illustration)

```
# Notebook titled <notebook title>
## Friday, June 30th, 2023 at 2:13 PM ET, Mr. Bold A. Busted wrote:

<content>

## Monday July 3rd, 2023 at 11:30 AM ET, Mr. Bold A. Busted wrote:

<content>
```

Is this already possible? Thanks again for nb! :) Really loving it after discovering it last week! :)

**Comment by boldandbusted** [comment]
Should I create a "Feature Request" in Issues for this? :) This one feature would make nb just perfect for my present use-case. If there's some quick scripts I should consider creating (like a python code and relevant modules), I'm open to that, too. (And maybe those could go into a "contrib" subdirectory?)

**Comment by xwmx** [comment]
@boldandbusted Glad you like it! This could be implemented as a [plugin](https://github.com/xwmx/nb#-plugins). You could just iterate over each file in the current or target notebook, write to a big markdown file, and use `pandoc` for the conversions. No need to create a feature request. I can envision how it could look in shell and can try to implement it at some point if you haven't already by then.

**Reply by boldandbusted** [reply]
@xwmx I'll have to dig into the plugin system a bit more - it looks straightforward, but... it is as simple as the *NIX shell! ;)
